subnet/kube/kube.go

```diff
@@ -0,0 +1,292 @@
+		return nil, fmt.Errorf("failed to write cni conf: %v", err)
+	}
+
+	sc, err := subnet.ParseConfig(string(netConf))
+	if err != nil {
+		return nil, fmt.Errorf("error parsing subnet config: %s", err)
+	}
+	sm, err := newKubeSubnetManager(c, sc, nodeName)
+	if err != nil {
+		return nil, fmt.Errorf("error creating network manager: %s", err)
+	}
+	go sm.Run(context.Background())
+	return sm, err
+}
+
+func newKubeSubnetManager(c clientset.Interface, sc *subnet.Config, nodeName string) (*kubeSubnetManager, error) {
+	var ksm kubeSubnetManager
+	ksm.client = c
+	ksm.nodeName = nodeName
+	ksm.subnetConf = sc
+	ksm.nodeStore.Store, ksm.nodeController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return ksm.client.Core().Nodes().List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return ksm.client.Core().Nodes().Watch(options)
+			},
+		},
+		&api.Node{},
+		resyncPeriod,
+		framework.ResourceEventHandlerFuncs{},
+	)
+	return &ksm, nil
+}
+
+func (ksm *kubeSubnetManager) GetNetworkConfig(ctx context.Context, network string) (*subnet.Config, error) {
+	time.Sleep(time.Second)
+	return ksm.subnetConf, nil
+}
+
+func (ksm *kubeSubnetManager) AcquireLease(ctx context.Context, network string, attrs *subnet.LeaseAttrs) (*subnet.Lease, error) {
+	time.Sleep(time.Second)
+	nobj, found, err := ksm.nodeStore.Store.GetByKey(ksm.nodeName)
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, fmt.Errorf("node %q not found", ksm.nodeName)
+	}
+	n, ok := nobj.(*api.Node)
+	if !ok {
+		return nil, fmt.Errorf("nobj was not a *api.Node")
+	}
+	if n.Spec.PodCIDR == "" {
+		return nil, fmt.Errorf("node %q pod cidr not assigned", ksm.nodeName)
+	}
+	bd, err := attrs.BackendData.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	_, cidr, err := net.ParseCIDR(n.Spec.PodCIDR)
+	if err != nil {
+		return nil, err
+	}
+	if n.Annotations[backendDataAnnotation] != string(bd) ||
+		n.Annotations[backendTypeAnnotation] != attrs.BackendType ||
+		n.Annotations[backendPublicIPAnnotation] != attrs.PublicIP.String() ||
+		n.Annotations[subnetKubeManagedAnnotation] != "true" {
+		n.Annotations[backendTypeAnnotation] = attrs.BackendType
+		n.Annotations[backendDataAnnotation] = string(bd)
+		n.Annotations[backendPublicIPAnnotation] = attrs.PublicIP.String()
+		n.Annotations[subnetKubeManagedAnnotation] = "true"
+		n, err = ksm.client.Core().Nodes().Update(n)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &subnet.Lease{
+		Subnet:     ip.FromIPNet(cidr),
+		Attrs:      *attrs,
+		Expiration: time.Now().Add(24 * time.Hour),
+	}, nil
+}
+
+func (ksm *kubeSubnetManager) RenewLease(ctx context.Context, network string, lease *subnet.Lease) error {
+	return nil
+}
+
+func (ksm *kubeSubnetManager) WatchLease(ctx context.Context, network string, sn ip.IP4Net, cursor interface{}) (subnet.LeaseWatchResult, error) {
+	time.Sleep(time.Second)
```

Comment:
What's this method called by? Would it make sense to to only sleep when an error indicates that the _system_ is not ready yet?